src/lib/reasoners/bitv_rel.ml

```diff
@@ -1489,6 +1522,13 @@ end = struct
       acts.acts_add_eq ~ex X.top X.bot;
       true
     | Rule | Rugt -> false
+    | Rdistinct when X.equal x y ->
+      acts.acts_add_eq ~ex X.top X.bot;
+      true
+    | Rdistinct ->
+      (* TODO(bclement): Is it is OK to simplify here if either side is
+         constant? *)
```

Comment:
What simplifications did you have in your mind?

Reply:
I mean remove the constraint entirely if we already know that the value is impossible.

Reply:
And you think it could be unsound? I agree to postpone this improvement, I just want to understand what you wanted to do here.

Reply:
No, there is no soundness risk (there might be incompleteness though).

Currently, we don't have access to the domains during simplification (only to the semantic values), so we can't directly check whether the value is possible or not. However, simplification should only run on constraints that were already propagated, so if one of the argument is a constant, we should be guaranteed that it is fine to drop the constraint in this case. I am not 100% certain it is a guarantee (there could be a bug or an unexpected interaction somewhere), so I don't want to depend on this before thoroughly checking that it is OK to do so.